flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java

```diff
@@ -102,74 +95,56 @@ public static String generatePlan(
-                    for (int inputNum = 0; inputNum < inputs.size(); inputNum++) {
-                        JobEdge edge = inputs.get(inputNum);
+                    for (int inputNum = 0; inputNum < vertex.getInputs().size(); inputNum++) {
+                        JobEdge edge = vertex.getInputs().get(inputNum);
                         if (edge.getSource() == null) {
                             continue;
                         }
-
                         JobVertex predecessor = edge.getSource().getProducer();
+                        if (predecessor == null || predecessor.getID() == null) {
+                            continue;
+                        }
+                        String inputId = predecessor.getID().toString();
 
+                        if (edge.getSource().getResultType() == null
+                                || edge.getSource().getResultType().name() == null) {
+                            continue;
+                        }
+                        String exchange = edge.getSource().getResultType().name().toLowerCase();
                         String shipStrategy = edge.getShipStrategyName();
                         String preProcessingOperation = edge.getPreProcessingOperationName();
                         String operatorLevelCaching = edge.getOperatorLevelCachingDescription();
 
-                        gen.writeStartObject();
-                        gen.writeNumberField("num", inputNum);
-                        gen.writeStringField("id", predecessor.getID().toString());
-
-                        if (shipStrategy != null) {
-                            gen.writeStringField("ship_strategy", shipStrategy);
-                        }
-                        if (preProcessingOperation != null) {
-                            gen.writeStringField("local_strategy", preProcessingOperation);
-                        }
-                        if (operatorLevelCaching != null) {
-                            gen.writeStringField("caching", operatorLevelCaching);
-                        }
-
-                        gen.writeStringField(
-                                "exchange", edge.getSource().getResultType().name().toLowerCase());
-
-                        gen.writeEndObject();
+                        inputs.add(
+                                new JobPlanInfo.Plan.Node.Input(
+                                        inputId,
+                                        inputNum,
+                                        exchange,
+                                        shipStrategy,
+                                        preProcessingOperation,
+                                        operatorLevelCaching));
                     }
-
-                    gen.writeEndArray();
                 }
-
-                // write the optimizer properties
-                gen.writeFieldName("optimizer_properties");
-                gen.writeRawValue(optimizerProps);
```

Comment:
Noticed `optimizerProps` is a string but is written as raw json - I should probably replicate that behavior
Will be addressed in a followup pr

Reply:
https://github.com/apache/flink/pull/26488